TutorLizard.BusinessLogic/Services/StudentService.cs

```diff
@@ -17,26 +17,70 @@ public AdRequest CreateAdRequest(int adId, string message)
     {
         // return created request
         // return empty with id = 0 to indicate something went wrong
+        int? studentId = _userIdentityService.GetUserId();
+        
+        if (studentId is null) 
+        {
+            return new AdRequest()
+            { AdId = 0 };
+        }
+
+        return _dataAccess.CreateAdRequest(adId, studentId.Value, isAccepted: false, message);
```

Comment:
Instead of `Value` cast studentId to int, like you did in `TutorService` when creating an Ad. That way when you ask VS to create a method based on this, it won't get confused about the type or the name of the parameter (value is a name that tells next to nothing).